server/server.go

```diff
@@ -75,6 +84,21 @@ func (s *Server) Handler(r *http.Request) http.Handler {
 	return h
 }
 
+func (s *Server) redirectOauth(w http.ResponseWriter, req *http.Request) {
+	// Shallow copy the existing URL
+	newDest := req.URL
+	// Replace the hostname with the provided hostname
+	newDest.Host = s.OauthRedirectURL.Host
```

Comment:
For completeness you may want to also include the port from `OauthRedirectUR` on `newDest`

Reply:
I don't follow.  The url.URL.Host field is both the hostname and the port, if any exists, so there's no extra field to copy.

Reply:
Ah, I always use `Hostname()` Never mind me.